src/lib/platform/exit_code.h

```diff
@@ -101,14 +102,16 @@ const char *otExitCodeToString(uint8_t aExitCode);
  * @param[in]   aExitCode   The exit code.
  *
  */
-#define VerifyOrDie(aCondition, aExitCode)                                                                   \
-    do                                                                                                       \
-    {                                                                                                        \
-        if (!(aCondition))                                                                                   \
-        {                                                                                                    \
-            otLogCritPlat("%s() at %s:%d: %s", __func__, __FILE__, __LINE__, otExitCodeToString(aExitCode)); \
-            exit(aExitCode);                                                                                 \
-        }                                                                                                    \
+#define VerifyOrDie(aCondition, aExitCode)                                                            \
+    do                                                                                                \
+    {                                                                                                 \
+        if (!(aCondition))                                                                            \
+        {                                                                                             \
+            otLogCritPlat("%s() at %s:%d: %s", __func__,                                              \
+                          (strrchr(__FILE__, '/') ? strrchr(__FILE__, '/') + 1 : __FILE__), __LINE__, \
```

Comment:
Is it possible to implement using templates in compile time?

Reply:
BTW, there is a `basename` syscall: https://man7.org/linux/man-pages/man3/basename.3.html

Reply:
`basename` is also runtime but more expense. I'm not aware of a template that can retrieve the basename..

Reply:
Do we really care about performance when the programming is quitting soon?
BTW, how do we know `basename` performs worse than the current code because we will actually call `strrchr` twice?

Reply:
We can consider adding a `constexpr` template function which does something similar `strrchr`, something similar to suggestion here: https://stackoverflow.com/a/54335644. 
The advantage is it would be all done at compile time.

Reply:
This is a c header. We cannot use templates here.

Reply:
I think using `strrchr` should be good enough. No need to worry about performance since the process is quitting right away.

Reply:
I fully agree that `strrchr` makes sense in this case.